2021/Robot2021/src/main/cpp/subsystems/Drivetrain.cpp

```diff
@@ -242,46 +247,34 @@ void Drivetrain::TalonFollowerInitialize(WPI_BaseMotorController &motor, int mas
 //
 void Drivetrain::UpdateOdometry(void)
 {
-    m_odometry.Update(
-        frc::Rotation2d(radian_t(GetHeadingAngle())),
-        GetDistanceMetersLeft(),
-        GetDistanceMetersRight());
+    // Get all sensors and update odometry
+    m_distanceLeft = GetDistanceMetersLeft();
+    m_distanceRight = GetDistanceMetersRight();

```

Comment:
Just out of curiosity, what was the point of creating separate variables for GetDistanceMetersLeft and Right?

Reply:
We previously had m_encoderLeft and m_encoderRight which held the Talon encoder counts. Then anywhere we needed distance we would need to calculate that each time. So part of the change was so that we would be working with "distance" everywhere instead of encoder counts.

But the advantage of holding these values in a variable is that we can read the sensor (look inside GetDistanceMetersLeft/Right) and everyone else gets access to this information without re-reading the actual hardware sensor. If there is a cost (processing time or delay) in reading that hardware (which there was a couple years ago on the CAN bus), then calling GetDistanceMetersLeft/Right will happen everywhere in the code it's needed and so will that delay.

We read the distance, velocity, and gyro angle once every periodic loop and store the value for everyone else to use. Being stored in RAM in that variable has virtually zero delay or cost relative to the processor as compared to a hardware access.

It's a good question, and a subtle reason unless you've run into hardware that is costly to read. Welcome to robot programming as opposed to more traditional CS. :)